lib/widgets/dialog.dart

```diff
@@ -26,14 +26,16 @@ Future<void> showErrorDialog({
   final zulipLocalizations = ZulipLocalizations.of(context);
   return showDialog(
     context: context,
-    builder: (BuildContext context) => AlertDialog(
-      title: Text(title),
-      content: message != null ? SingleChildScrollView(child: Text(message)) : null,
-      actions: [
-        TextButton(
-          onPressed: () => Navigator.pop(context),
-          child: _dialogActionText(zulipLocalizations.errorDialogContinue)),
-      ]));
+    builder: (BuildContext context) => SingleChildScrollView(
+      child: AlertDialog(
```

Comment:
Can you help me understand more specifically what problem this is solving?

It looks like we're already doing [what is recommended](https://api.flutter.dev/flutter/material/AlertDialog-class.html#alert-dialogs-and-scrolling) "[i]f the content is too large to fit on the screen vertically":

> Consider using a scrolling widget for [content](https://api.flutter.dev/flutter/material/AlertDialog/content.html), such as [SingleChildScrollView](https://api.flutter.dev/flutter/widgets/SingleChildScrollView-class.html), to avoid overflow.

See that we use a `SingleChildScrollView` when we have anything to pass for `content` at all:

```dart
/// Displays an [AlertDialog] with a dismiss button.
///
/// Returns a [Future] that resolves when the dialog is closed.
Future<void> showErrorDialog({
  required BuildContext context,
  required String title,
  String? message,
}) {
  final zulipLocalizations = ZulipLocalizations.of(context);
  return showDialog(
    context: context,
    builder: (BuildContext context) => AlertDialog(
      title: Text(title),
      content: message != null ? SingleChildScrollView(child: Text(message)) : null,
      actions: [
        TextButton(
          onPressed: () => Navigator.pop(context),
          child: _dialogActionText(zulipLocalizations.errorDialogContinue)),
      ]));
}
```

Are we not following the documentation correctly, or is there some different issue not covered there?

Reply:
There's a different issue noticed, this was noticed when the app is on landscape and the TextField of AddAccountPage has focus, with focus set on the TextField, the keyboard is out which takes up most of the space on the screen, then, there is a noticeable overflow when the dialog pops up to show an error, after the addaccount button is clicked, this is not about the content but the size of the dialog, which does  not fit to the screen I used in testing, which is  an Android Emulator that emulates a Google Pixel4 device, with a resolution(dp) 393x830 and a density 440dpi. As seen in the short before and after videos below.

https://github.com/user-attachments/assets/9638efad-a30b-4324-8680-1e12da370e74


https://github.com/user-attachments/assets/03a975ec-49ea-4450-98c4-3ff8a1773639

Reply:
Hmm. It's a good thing that the keyboard goes away, otherwise the dialog wouldn't have enough room to display anything in a reasonable way.

In the moment before the keyboard slides away, it looks like the "Invalid input" element—the dialog's title—is what's overflowing.

Instead of putting the whole dialog in a `SingleChildScrollView`, I'd like to try a different approach that still uses `AlertDialog`'s documented API:

1. Remove the `SingleChildScrollView` in the `content` field:
   
   ```diff
   -      content: message != null ? SingleChildScrollView(child: Text(message)) : null,
   +      content: message != null ? Text(message) : null,
   ```
2. Pass [`scrollable: true`](https://api.flutter.dev/flutter/material/AlertDialog/scrollable.html) to the `AlertDialog` constructor

From reading the documentation, this should make it so the title is included in the scrollable too. Does this fix the overflow issue? If so, please put (1) and (2) in the same commit, applied to both `showErrorDialog` and `showSuggestedActionDialog`.

Reply:
This fixes the overflow issue. I have applied it on `showErrorDialog` and `showSuggestedActionDialog`.

Reply:
Great! Please put the dialog-related changes in their own commit, separate from the home-screen and login-screen changes, and post on the PR thread when the PR is ready for another review.